src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -103,7 +103,12 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 		})
 	}
 
-	func showTestResultScreen(triggeredFromTeletan: Bool = false) {
+	func showTestResultScreen(triggeredFromTeletan: Bool = false, saveToContactDiary: Bool = false) {
+		if saveToContactDiary {
+			model.coronaTestService.createCoronaTestEntryInContactDiary(
```

Comment:
Shouldn't the diary entry for negative tests already be there without the need for the user to even open the test result screen? I think the original bug report was only meant for positive tests as we don't directly show the result

Reply:
Yes then we can save directly as soon as we get the test result, i will update the code

Reply:
@30mar  Would a negative test result now be safe twice?

Reply:
@ArturFriesen  no i removed the logic for the negative test from the coordinator, so it will be saved once in the service